content/docs/learning-guide/tutorials/environments/constructing-terrain.md

```diff
@@ -0,0 +1,96 @@
+3. In the **Entity Inspector**, add both the **Terrain World** and **Terrain World Renderer** components.
+- ![B](/images/learning-guide/tutorials/environments/constructing-terrain/B.png)
+
+### **Define a spawnable area**
+
+1. Create a new entity (hotkey **Ctrl+Alt+N**). (This will be your **TerrainSpawner** entity.)
+- ![C](/images/learning-guide/tutorials/environments/constructing-terrain/C.png)
+2. With the new entity selected, in the Entity Inspector, add a **Terrain Layer Spawner** and an **Axis Aligned Box Shape** component. (You can adjust the dimensions of the Aligned Box Shape component to encompass or intersect with the Terrain Layer Spawner.)
+3. Next, add a **Terrain Heigh Gradient List** component.
+- ![D](/images/learning-guide/tutorials/environments/constructing-terrain/D.png)
+4. In the Entity Outliner, create a second entity. (This will contain your **HeightfieldGradient** entity.)
+- ![E](/images/learning-guide/tutorials/environments/constructing-terrain/E.png)
+5. With the new entity selected, in the Entity Inspector, add an **Image Gradient** Component.
+6. Next, add the required **Gradient Transfer Modifier** component.
+7. Then, add a required Shape component (e.g. **Box Shape, Sphere, Shape, Axis Aligned Box Shape**).
+    Typically, this will be a Shape Reference so that we can share the same shape as the Terrain Layer Spawner, so that the image always exactly stretches across the entire spawner.
+ -  ![F](/images/learning-guide/tutorials/environments/constructing-terrain/F.png)
+8. In the Shape Reference component (if used), assign the TerrainSpawner entity.
+9. In the TerrainSpawner entity, assign the HeightfieldGradient entity within the Terrain Height Gradient List component.
+- ![G](/images/learning-guide/tutorials/environments/constructing-terrain/G.png)
+10. Now you should see constructed terrain using a heightfield gradient.
+-  ![H](/images/learning-guide/tutorials/environments/constructing-terrain/H.png)
+11. **Save** the level (hotkey **Ctrl+S**).
+
+## **Applying Materials to Terrain**
+
+You can generate terrain in a few different ways via the Entity Outliner and Entity Inspector.
+
+**Setup**: See Constructing Terrain.
+
+### **Create a Macro Material**
+
+1. In the Entity Outliner, select the TerrainSpawner entity.
+- ![I](/images/learning-guide/tutorials/environments/constructing-terrain/I.png)
+2. In the Entity Inspector, add a **Terrain Macro Material** component.
+- ![J](/images/learning-guide/tutorials/environments/constructing-terrain/J.png)
+3. In the Terrain Macro Material component, select a **Color Texture** and a **Normal Texture** from the **Pick Streamingimage** modal.
+- ![K](/images/learning-guide/tutorials/environments/constructing-terrain/K.png)
+4. **Save** the level (hotkey **Ctrl+S**).
+
+*Note: You can adjust normals as desired within the Terrain Macro Material component.*
+
+### **Assign Detail Materials**
```

Comment:
There's a section missing here describing how to Assign Surface Types. The steps for it are roughly as follows:
* In the Asset Editor, create a new "Surface Tag Name List" asset, fill it with all of the surface names you'd like to use, and save the asset in your level or project.
* Add a Terrain Surface Gradient List component.
* For each surface type, create a new Image Gradient entity (similar to the steps for the Height Image Gradient) that contains the weights for that surface.
* Add each of those entities to the Terrain Surface Gradient List and assign them the desired surface name.

These surface names are now the ones that are available for the Assign Detail Materials section, where render materials can be matched up with the surfaces.